internal/fnruntime/container.go

```diff
@@ -157,27 +157,18 @@ func newContainerEnvFromStringSlice(envStr []string) *runtimeutil.ContainerEnv {
 // prepareImage will check local images and pull it if it doesn't
 // exist.
 func (f *ContainerFn) prepareImage() error {
-	// check image existence
-	args := []string{"image", "ls", f.Image}
-	cmd := exec.Command(dockerBin, args...)
-	var output []byte
-	var err error
-	if output, err = cmd.CombinedOutput(); err != nil {
-		return fmt.Errorf("failed to check local function image %q: %w", f.Image, err)
-	}
-	if strings.Contains(string(output), strings.Split(f.Image, ":")[0]) {
-		// image exists locally
-		return nil
-	}
-	args = []string{"image", "pull", f.Image}
+	// We always try to pull the image regardless if the tag has been seen in
+	// the local cache. Because we want to ensure we have the latest release for
+	// tags like v1 and v1.2 that are moving tags.
+	args := []string{"image", "pull", f.Image}
 	// setup timeout
 	timeout := defaultTimeout
 	if f.Timeout != 0 {
 		timeout = f.Timeout
 	}
 	ctx, cancel := context.WithTimeout(context.Background(), timeout)
 	defer cancel()
-	cmd = exec.CommandContext(ctx, dockerBin, args...)
+	cmd := exec.CommandContext(ctx, dockerBin, args...)
```

Comment:
So in this approach, we no longer trigger fetching of images in parallel.

Reply:
Right, let's ensure the correctness first.
We can enhance the performace later in https://github.com/GoogleContainerTools/kpt/issues/2026

Reply:
Ack.